### ISSUE TITLE
Custom completion rule completionpassgroup of module mumie has wrong suffix

### ISSUE THREAD

**Issue by voegii** [integral-learning/moodle-mod_mumie#76]
On course default completion (/course/defaultcompletion.php) following error occurs:

```
Custom completion rule completionpassgroup of module mumie has wrong suffix and has been removed from the form. This has to be fixed by the developer
line 128 of /completion/classes/edit_base_form.php: call to debugging()
line 153 of /completion/classes/edit_base_form.php: call to core_completion_edit_base_form->add_custom_completion()
line 178 of /completion/classes/form/form_trait.php: call to core_completion_edit_base_form->add_completion_rules()
line 249 of /completion/classes/edit_base_form.php: call to core_completion_edit_base_form->add_completion_elements()
line 102 of /completion/classes/defaultedit_form.php: call to core_completion_edit_base_form->definition()
line 217 of /lib/formslib.php: call to core_completion_defaultedit_form->definition()
line 47 of /completion/classes/defaultedit_form.php: call to moodleform->__construct()
line 66 of /course/classes/output/bulk_activity_completion_renderer.php: call to core_completion_defaultedit_form->__construct()
line 97 of /course/defaultcompletion.php: call to core_course_bulk_activity_completion_renderer->defaultcompletion()
```

Seen on Moodle 4.4, most probably also on Moodle 4.3

**Comment by Lyannic** [comment]
Thanks for the report!

Apparently, since Moodle 4.3, you need to add a suffix to the completion rules: https://moodledev.io/docs/4.3/devupdate#activity-completion.
It suggests using `$this->get_suffix()` from `completion/classes/form/form_trait.php` which is not available prior to 4.3. So we need to add a check for the Moodle version and not add the suffix for Moodle versions < 4.3.
Taking inspiration from https://github.com/ndunand/moodle-mod_choicegroup/pull/201/files, this could be done with `$CFG->branch`
```php
if ($CFG->branch < 403) {
    $suffix = '';
} else {
    $suffix = $this->get_suffix();
}
```

**Comment by Lyannic** [comment]
Bug is fixed an released: https://github.com/integral-learning/moodle-mod_mumie/releases/tag/v1.7.1.
